scio-core/src/main/scala/com/spotify/scio/values/SCollection.scala

```diff
@@ -972,18 +972,7 @@ sealed trait SCollection[T] extends PCollectionWrapper[T] {
    */
   def saveAsTextFile(textIO: TextIO)(implicit writeP: textIO.WriteParams = textIO.WriteParams())
   : Future[Tap[String]] = {
-    val s = if (classOf[String] isAssignableFrom this.ct.runtimeClass) {
-      this.asInstanceOf[SCollection[String]]
-    } else {
-      this.map(_.toString)
-    }
-
-    if (context.isTest) {
-      context.testOut(textIO)(s)
-      s.saveAsInMemoryTap
-    } else {
-      textIO.write(s, writeP)
-    }
+   write(textIO)(writeP)
```

Comment:
This gives me Type inference error, tried couple of ways to resolve that but it seems type information of the higher order type (type T in ScioIO[T]) get removed while passing down to the actual write method.

@jto would be great if you can have a look at this error.

Reply:
Does forcing the type works ? `write[ExpectedType](textIO)(writeP)`

Reply:
Tried but this gives me shadowing type compilation error as it shadows the type `T` in `SCollection[T]`

Reply:
I think I understood the root cause, working on a fix.